[Feature] Rgcn integration

xiao03 opened the pull request:
## Description
<!-- Brief description. Refer to the related issues if existed.
It'll be great if relevant reviewers can be assigned as well.-->

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../README.md).

## Changes
<!-- You could use following template
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)
-->

xiao03 committed:
initial merge, awaiting test

xiao03 committed:
add library code and test

hugochan committed:
add rgcn for QG

hugochan committed:
add rgcn for text classification

xiao03 committed:
update config

xiao03 committed:
format the script

xiao03 committed:
remove device options from model

AlanSwift commented on graph4nlp/pytorch/modules/graph_embedding_learning/rgcn.py:
```diff
@@ -0,0 +1,249 @@
+        num_bases=None,
+        use_self_loop=True,
+        dropout=0.0,
+    ):
```
1. You have to support the "direction_option" since it is an important feature. For more details, please refer to the implementation of [GCN](https://github.com/graph4ai/graph4nlp/blob/c20b9c227b2253c8e316d7794091f8e469ff10e1/graph4nlp/pytorch/modules/graph_embedding_learning/gcn.py) or ask me directly.
2. Please keep the arguments consistent with other GNN implementations. E.g., dropout --> feat_drop. Reference: [GCN's arguments](https://github.com/graph4ai/graph4nlp/blob/c20b9c227b2253c8e316d7794091f8e469ff10e1/graph4nlp/pytorch/modules/graph_embedding_learning/gcn.py#L66).
3.

AlanSwift replied:
resloved

AlanSwift commented on graph4nlp/pytorch/modules/graph_embedding_learning/rgcn.py:
```diff
@@ -0,0 +1,249 @@
+        activation=None,
+        self_loop=False,
+        dropout=0.0,
+        layer_norm=False,
```
More arguments should be exposed to users. E.g., regularizer, bias.

AlanSwift reviewed (changes requested):
Please refer to the detailed comments.

xiao03 committed:
add direction_options for rgcn

xiao03 committed:
format

xiao03 committed:
remove unused parameters

xiao03 committed:
isort fix

xiao03 committed:
isort fix again

AlanSwift committed:
fix ci

xiao03 committed:
update rgcn & test case

hugochan committed:
update readme

AlanSwift committed:
fix

xiao03 committed:
bug fix on bi sep

xiao03 committed:
bug fix on bi_fuse

SaizhuoWang committed:
Change implementation of RGCN linear layer to DGL impl

SaizhuoWang committed:
Merge branch 'add_rgnn_for_qg' of https://github.com/graph4ai/graph4nlp into rgcn-integration

SaizhuoWang committed:
Implemented regularizer in RGCN

SaizhuoWang committed:
Sync RGCNLayer implementation

SaizhuoWang committed:
modified DGL benchmark test code for rgcn

hugochan committed:
linter & update readme

hugochan committed:
switch to RGCNLayer implemented in https://github.com/graph4ai/graph4nlp/pull/584

SaizhuoWang committed:
Migrated RGCN layer to hetero version in DGL

SaizhuoWang committed:
fixed rgcn interface with dgl issue with tricks

SaizhuoWang committed:
Bugfix in GraphData and RGCN testing
- Disabled Valuerror for single node type for homograph node adding
- Fixed DGL benchmark test for RGCN

SaizhuoWang committed:
mlflow integration and test script
